### ISSUE TITLE
n+1 in Api::V1::VersionsController#show

### ISSUE THREAD

**Issue by sonalkr132** [rubygems/rubygems.org#1450]
`gem_download` association of versions model is instance dependent:

``` Ruby
 has_one :gem_download, proc { |m| where(rubygem_id: m.rubygem_id) }
```

download_count of versions payload uses this association causing n queries for download count of each version. Logically this should take only one query to load all the gem_downloads. I am not sure how can this be eager loaded. Perhaps with a decorator?

**Comment by ecnelises** [comment]
Since `gem_download` also belongs to `rubygem`, is there any need for a set of versions to query their respective download count? Or if we want to, putting them into a subquery like `GemDownload.where(version_id: versions.pluck(:id))` will reduce N+1 to 1.

Did I misunderstand anything?

**Comment by sonalkr132** [comment]
closed by #2086
